src/main/java/com/st/utopia/cancellationservice/controller/CancellationServiceController.java

```diff
@@ -0,0 +1,51 @@
+package com.st.utopia.cancellationservice.controller;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+import com.st.utopia.cancellationservice.model.Ticket;
+import com.st.utopia.cancellationservice.service.TicketService;
+
+import javassist.NotFoundException;
+
+
+
+@RestController
+@RequestMapping("/Cancel")
+public class CancellationServiceController {
+	
+	@Autowired
+	private TicketService ticketservice;
+	
+	
+	@GetMapping("/FindTicket/{flightNumber}/{rowNumber}/{seat}")
+	public ResponseEntity<Ticket> getTicket(@PathVariable int flightNumber, @PathVariable int rowNumber,@PathVariable char seat) throws NotFoundException {

```

Comment:
Does the cancellation service really need to have a get-ticket-by-seat-location route?

Reply:
Yes, you do need all of that because they make up the composite key. Unless he searches via booking_id

Reply:
I mean, does the user-facing-API middleware really need to get the details of a given ticket? The other routes this API exposes will all take the flight number, row, and seat anyway.